farm/data_handler/data_silo.py

```diff
@@ -93,29 +101,42 @@ def _get_dataset(self, filename, dicts=None):
                         random.shuffle(dicts)
 
         num_dicts = len(dicts)
-        multiprocessing_chunk_size, num_cpus_used = calc_chunksize(num_dicts, max_chunksize=self.max_multiprocessing_chunksize)
 
         with ExitStack() as stack:
-            p = stack.enter_context(mp.Pool(processes=num_cpus_used))
-
-            logger.info(
-                f"Got ya {num_cpus_used} parallel workers to convert {num_dicts} dictionaries "
-                f"to pytorch datasets (chunksize = {multiprocessing_chunk_size})..."
-            )
-            log_ascii_workers(num_cpus_used, logger)
-
-            results = p.imap(
-                partial(self._multiproc, processor=self.processor),
-                grouper(dicts, multiprocessing_chunk_size),
-                chunksize=1,
-            )
+            if self.max_processes > 1:  # use multiprocessing only when max_processes > 1
+                multiprocessing_chunk_size, num_cpus_used = calc_chunksize(
+                    num_dicts=num_dicts,
+                    max_processes=self.max_processes,
+                    max_chunksize=self.max_multiprocessing_chunksize,
+                )
+
+                p = stack.enter_context(mp.Pool(processes=num_cpus_used))
+
+                logger.info(
+                    f"Got ya {num_cpus_used} parallel workers to convert {num_dicts} dictionaries "
+                    f"to pytorch datasets (chunksize = {multiprocessing_chunk_size})..."
+                )
+                log_ascii_workers(num_cpus_used, logger)
+
+                results = p.imap(
+                    partial(self._dataset_from_chunk, processor=self.processor),
+                    grouper(dicts, multiprocessing_chunk_size),
+                    chunksize=1,
+                )
+            else:
+                logger.info(
+                    f"Multiprocessing disabled, using a single worker to convert {num_dicts}"
+                    f"dictionaries to pytorch datasets."
+                )
+
+                results = map(partial(self._dataset_from_chunk, processor=self.processor), grouper(dicts, num_dicts))
 
             datasets = []
-            with tqdm(total=len(dicts), unit=' Dicts') as pbar:
+            with tqdm(total=len(dicts), unit=" Dicts", desc="Converting Dicts to PyTorch Datasets") as pbar:
```

Comment:
Just added another description here earlier today: https://github.com/deepset-ai/FARM/pull/175/files
I would prefer "Preprocessing Dataset" here since dicts is not very clear for users who are not familiar with the FARM processors yet